src/serviceDesks/incontact/inContactServiceDesk.ts

```diff
@@ -91,22 +92,22 @@ class InContactServiceDesk implements ServiceDesk {
         body: JSON.stringify({}),
       });
       const output = await request.json();
-      if (output && output.chatSessionId) {
+      if (output.chatSessionId) {
         this.session = output;
       }
     } catch (error) {
-      return Promise.reject();
+      throw new Error(error);
     }
 
-    // The beforeunload event is fired when the window, the document and its resources are about to be unloaded.
+    // The before unload event is fired when the window, the document and its resources are about to be unloaded.
     window.addEventListener('unload', (event) => {
       navigator.sendBeacon(`${this.SERVER_BASE_URL}/incontact/end`, JSON.stringify(this.session)); // https://golb.hplar.ch/2018/09/beacon-api.html
     });
 
     try {
       await this.startPolling();
     } catch (error) {
-      return Promise.reject();
+      throw new Error(error);
```

Comment:
This `try...catch` is no longer necessary for two reasons. One is that `startPolling` won't actually throw any errors anymore. It now has a `try...catch` inside that catches everything and no more returning of rejected promises. Second it's not really necessary to catch an error and then wrap and re-throw it. If you delete the `try...catch` here and `startPolling` does throw an error it will simply get thrown out of this function which accomplishes the same thing as `throw new Error(error)`.

Reply:
Hmm... this makes a lot of sense but I kept this because otherwise the console logs would so an unhandled promise rejection error.

```
UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict`
```
Does that mean i have to add a `.catch()` to the when we invoke the function?

Reply:
I think this is ok because I was testing out some unreachable scenarios, and all the errors thrown by the InContact API calls itself are handled appropriately. Maybe would be problematic if the middleware server was bugging out though, or if there are errors in the middleware code, so this makes me want unit tests lol.